PR: Add FAQ questions on Spyder being open source and Anaconda situation

CAM-Gerlach opened the pull request:
<!--- Before submitting your pull request, --->
<!--- please complete as much as possible of the following checklist: --->

### Pull Request Checklist

* [x] Read and followed this repo's [Contributing Guidelines](https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md)
* [x] Based your PR on the latest version of the correct branch (master or 3.x)
* [x] Checked your writing carefully for correct English spelling, grammar, etc
* [x] Described your changes and the motivation for them below
* [x] Noted what issue(s) this pull request resolves, creating one if needed



## Description of Changes

<!--- Describe what you've changed and why. --->

Adds a new section, "About Spyder", with two new FAQ questions: one clarifying that Spyder is open source and fully free to use, one explaining how to use it without running in to the Anaconda EULA restrictions on large commercial companies.

![image](https://user-images.githubusercontent.com/17051931/101269081-3c9fde00-3730-11eb-9e91-cef3dc64f21a.png)



### Issue(s) Resolved

<!--- Pull requests should typically resolve at least one—preferably only one—
<!--- outstanding issue; create a new one if no relevant issue exists.
<!--- List the issue(s) below, in the form "Fixes #1234" . One per line.--->

Fixes #202 


<!--- Thanks for your help making Spyder --->
<!--- and its documentation better for everyone! --->

juanis2112 commented on doc/faq.rst:
```diff
@@ -42,6 +42,25 @@ Installing and updating
+
+   Spyder is 100% free and open source; there is no paid version or prohibition on commercial use.
+   Spyder is developed by its international user community, and supported by its users through `OpenCollective`_ and by its generous sponsoring organizations, including `Quansight`_ and `NumFOCUS`_.
+   Our source code, standalone installers and most of our distribution methods (Pip/PyPI, Linux distros, MacPorts, WinPython, etc) are fully unencumbered and can be freely redistributed, used and modified by anyone.
```
I would suggest replacing the "unencumbered" word for something more familiar (specially for non native speakers).

juanis2112 commented on doc/faq.rst:
```diff
@@ -42,6 +42,25 @@ Installing and updating
+
+   Spyder is 100% free and open source; there is no paid version or prohibition on commercial use.
+   Spyder is developed by its international user community, and supported by its users through `OpenCollective`_ and by its generous sponsoring organizations, including `Quansight`_ and `NumFOCUS`_.
+   Our source code, standalone installers and most of our distribution methods (Pip/PyPI, Linux distros, MacPorts, WinPython, etc) are fully unencumbered and can be freely redistributed, used and modified by anyone.
```
I would suggest replacing the "unencumbered" word for something more familiar (specially for non native speakers).

CAM-Gerlach replied:
```suggestion
   Our source code, standalone installers and most of our distribution methods (Pip/PyPI, Linux distros, MacPorts, WinPython, etc) can be freely redistributed, used and modified by anyone, for any purpose, including commercial use.
```

Yeah, I don't know why I thought that was a good idea in a public FAQ for beginners from around the world. Ended up removing it completely and instead describing what I meant more clearly and explicitly.

juanis2112 commented on doc/faq.rst:
```diff
@@ -42,6 +42,25 @@ Installing and updating
+   .. _Quansight: https://www.quansight.com/
+   .. _NumFOCUS: https://numfocus.org/
+
+   The Anaconda distribution recently `recently changed`_ their EULA to add restrictions on large (> 200 employee) for-profit companies using it on a large scale, but that only applies to the package distribution infrastructure (the packaged Anaconda distribution and the ``defaults`` conda channel).
```
```suggestion
   The Anaconda distribution recently `changed`_ their EULA to add restrictions on large (> 200 employee) for-profit companies using it on a large scale, but that only applies to the package distribution infrastructure (the packaged Anaconda distribution and the ``defaults`` conda channel).
```
Repeated word

CAM-Gerlach replied:
```suggestion
   The Anaconda distribution `recently changed`_ their EULA to add restrictions on large (> 200 employee) for-profit companies using it on a large scale, but that only applies to the package distribution infrastructure (the packaged Anaconda distribution and the ``defaults`` conda channel).
```

Avoid breaking the link reference

juanis2112 reviewed (changes requested):
I think it might be better to separate these two in two different questions in the FAQ as they are not actually too related and it makes the answer too long to have them like this. Also if we do this, we can create another section that can be called like "About Spyder" or something like that, since from the question the connection to the section is not very clear. 

I would also suggest, if we split these questions up, making them more specific and clear cause for instance the "What's the situation with Anaconda" one it's not very clear as in... what situation specifically you are talking about.

ccordoba12 commented on doc/faq.rst:
```diff
@@ -42,6 +42,25 @@ Installing and updating
+   .. _NumFOCUS: https://numfocus.org/
+
+   The Anaconda distribution recently `recently changed`_ their EULA to add restrictions on large (> 200 employee) for-profit companies using it on a large scale, but that only applies to the package distribution infrastructure (the packaged Anaconda distribution and the ``defaults`` conda channel).
+   Instead, you can simply download the similar `Miniconda distribution`_, which is 100% open source and identical to full Anaconda except for the gigabytes of Python packages installed by default in the Anaconda ``base`` environment (which we recommend you avoid using anyway).
```
```suggestion
   Instead, you can simply download the similar `Miniconda distribution`_, which is 100% open source and identical to full Anaconda except for the many Python packages installed by default in the Anaconda ``base`` environment (which we recommend you avoid using anyway).
```

- The phrasing seems derogatory, which reads opinionated and not something I think we should include in our FAQ.
- Also, perhaps we should point directly to Miniforge instead of Miniconda?

CAM-Gerlach replied:
> The phrasing seems derogatory, which reads opinionated and not something I think we should include in our FAQ.

Agreed

> Also, perhaps we should point directly to Miniforge instead of Miniconda?

Absolutely, didn't even know that existed. I'll do it when I pull and rebase these in, since I'll need to change the link target as well

CAM-Gerlach replied:
Done; also modified the follow sentences to account for that change.

ccordoba12 commented on doc/faq.rst:
```diff
@@ -42,6 +42,25 @@ Installing and updating
+
+   The Anaconda distribution recently `recently changed`_ their EULA to add restrictions on large (> 200 employee) for-profit companies using it on a large scale, but that only applies to the package distribution infrastructure (the packaged Anaconda distribution and the ``defaults`` conda channel).
+   Instead, you can simply download the similar `Miniconda distribution`_, which is 100% open source and identical to full Anaconda except for the gigabytes of Python packages installed by default in the Anaconda ``base`` environment (which we recommend you avoid using anyway).
+   Then, simply install whatever packages you need (including Spyder, if you aren't using our recommended stand-alone installers) from the community-maintained ``conda-forge`` channel, which has a much wider variety of packages and is generally more up to date than the Anaconda ``defaults`` channel.
```
```suggestion
   Then, simply install the packages you need (including Spyder, if you aren't using our recommended stand-alone installers) from the community-maintained ``conda-forge`` channel, which has a much wider variety of packages and is generally more up to date than the Anaconda ``defaults`` channel.
```

- I don't like `whatever` too much here.
- I think we should also add a link to the section in our Installation guide about our new installers.

CAM-Gerlach replied:
> I don't like whatever too much here.

Agreed

> I think we should also add a link to the section in our Installation guide about our new installers.

Good point, I'll do that in the rebase as well

CAM-Gerlach replied:
Added

ccordoba12 reviewed (commented):
Besides my suggestions below, I also agree with @juanis2112's very good suggestions: we should split the text in two questions and add a new secton for them.

My suggestion for the header of that section: `Spyder and Anaconda`

CAM-Gerlach commented:
@ccordoba12 @juanis2112 Rebased in the links, added the class ids, applied your changes, made the additional changes @ccordoba12 suggested, split the questions and added them to a new section as @juanis2112 , and fixed/improved the wording and formatting in a number of places.

ccordoba12 commented:
@CAM-Gerlach, this has a merge conflict now.

CAM-Gerlach commented:
@ccordoba12 Rebased to fix.

CAM-Gerlach commented:
@ccordoba12 @juanis2112 Bump

juanis2112 commented:
Will review now, sorry

juanis2112 commented on doc/faq.rst:
```diff
@@ -321,6 +321,40 @@ Troubleshooting
+   :container: + dropdown-id-anaconda-license
+
+   If you use Spyder with the Anaconda distribution, they `recently changed`_ their `Terms of Service`_ to add restrictions on large (> 200 employee) for-profit companies using Anaconda on a large scale.
+   However, these terms only applies to the package infrastructure (the full Anaconda distribution and the ``defaults`` conda channel).
```
```suggestion
   However, these terms only apply to the package infrastructure (the full Anaconda distribution and the ``defaults`` conda channel).
```
Isn't it like this?

CAM-Gerlach replied:
Yes indeed, good catch. I'll fixup it in.

CAM-Gerlach replied:
Done

juanis2112 reviewed (approved):
I think this looks good now

ccordoba12 reviewed (approved):
It looks pretty good @CAM-Gerlach! Thanks for your work on this.
